junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/TestFactoryTestDescriptor.java

```diff
@@ -69,6 +70,18 @@ public boolean isLeaf() {
 	}
 
 	@Override
+	public JupiterEngineExecutionContext execute(JupiterEngineExecutionContext context,
```

Comment:
I choose to override the `execute()` method here to prevent calls to _before_ and _after_ hooks around a container execution.
This is because a hook implementation can not know on which (of a container or a test) it is executed.
I assumed that `@BeforeEach` meant before each test, in this particular case, unlike what's said in the annotations javadoc.


A cleaner way can be to declare a new annotation `@BeforeEachTest` and let callback implementations know about the type of test (container or not).

If this is something you prefer, I can give it a shot.